src/cluster_legacy.c

```diff
@@ -6890,9 +6939,26 @@ void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
 
+void updateAllCachedNodesHealth(void) {
+    dictIterator *di =  zmalloc(sizeof(*di));
+    dictEntry *de;
+    clusterNode *node;
+    int overall_health_changed = 0;
+    dictInitSafeIterator(di, server.cluster->nodes);
+    while((de = dictNext(di)) != NULL) {
```

Comment:
```suggestion
    dictIterator di;
    dictInitSafeIterator(&di, server.cluster->nodes);
    dictEntry *de;
    clusterNode *node;
    int overall_health_changed = 0;
    while((de = dictNext(&di)) != NULL) {
```
This is how you use the stack allocated version.

Reply:
Oh! Thank you so much!

Reply:
I tried this, but thought would be a wrong way so I changed it again